### PR TITLE
Fix indexing and versioning issues

### PR DESCRIPTION
**Pull request by pazembrz**
#1559

### CODE REVIEW

File: backend/inspirehep/records/api/base.py [hunk 1]
```diff
@@ -410,21 +460,29 @@ def update(self, data, *args, **kwargs):
...
+                else:
+                    self.delete_records_from_deleted_records(data)
+                    self.pidstore_handler.update(self.id, self)
+                    if self.get("deleted_records"):
+                        self.redirect_pids(self["deleted_records"])
+                self.update_model_created_with_legacy_creation_date()
+                flag_modified(self.model, "json")
+                db.session.merge(self.model)
+                db.session.flush()
+                # After flush new version should be created.
```
Comment (drjova) [inline backend/inspirehep/records/api/base.py:480]: could we add a feature flag for this?

Comment (pazembrz) [reply backend/inspirehep/records/api/base.py:480]: Ah, yes. Right. I forgot about It...

File: backend/tests/integration-async/records/indexer/test_jobs.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        current_search.flush_and_refresh("records-jobs")
+        expected_records_count = 0
+        record_lit_es = JobsSearch().get_record(str(record.id)).execute().hits
+        assert expected_records_count == len(record_lit_es)
+
+    record = JobsRecord.create(faker.record("job"))
+    db.session.commit()
+
+    record.delete()
+    record.index(delay=False)
```
Comment (drjova) [inline backend/tests/integration-async/records/indexer/test_jobs.py:27]: why not commit as the rest of the tests :)

**Review by drjova — COMMENTED** [review]
**@pazembrz looks good 🥇 Just the comment and please squash the commits :)

File: backend/tests/integration-async/records/indexer/test_jobs.py (hunk 2 above)
Comment (pazembrz) [reply backend/tests/integration-async/records/indexer/test_jobs.py:27]: This one was moved directly from integration. And in integration we don't commit. I will update it accordingly to be consistent with other async tests.

**Review by drjova — APPROVED** [review]

**Pull request merged**